redfish-core/lib/cable.hpp

```diff
@@ -0,0 +1,198 @@
+#pragma once
+#include <boost/container/flat_map.hpp>
+#include <utils/json_utils.hpp>
+
+namespace redfish
+{
+using ManagedPropertyType =
+    boost::container::flat_map<std::string, std::variant<std::string, double>>;
+
+/**
+ * @brief Get Cable properties
+ * @param[in,out]   asyncResp   Async HTTP response.
+ * @param[in]       resp        A map to hold Cable properties
+ * @param[in]       cableId     Cable Id
+ */
+inline void
+    getCableProperties(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                       ManagedPropertyType& resp, const std::string& cableId)
+{
+    double* cableLength = nullptr;
+    std::string* cableTypeDescription = nullptr;
+
+    for (auto& propertyMap : resp)
+    {
+        if (propertyMap.first == "CableTypeDescription")
+        {
+            cableTypeDescription =
+                std::get_if<std::string>(&propertyMap.second);
+        }
+        else if (propertyMap.first == "Length")
+        {
+            cableLength = std::get_if<double>(&propertyMap.second);
+        }
+    }
+
+    if (cableTypeDescription == nullptr || cableLength == nullptr)
+    {
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    std::string dataId = "/redfish/v1/Cables/" + cableId;
+    asyncResp->res.jsonValue["@odata.type"] = "#Cable.v1_0_0.Cable";
+    asyncResp->res.jsonValue["@odata.id"] = dataId;
+    asyncResp->res.jsonValue["Id"] = cableId;
+    asyncResp->res.jsonValue["Name"] = "Cable Id";
+    asyncResp->res.jsonValue["LengthMeters"] = *cableLength;
+    asyncResp->res.jsonValue["CableType"] = *cableTypeDescription;
+}
+
+/**
+ * The Cable schema
+ */
+inline void requestRoutesCable(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Cables/<str>/")
+        .privileges({{"Login"}})
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& cableId) {
+                BMCWEB_LOG_DEBUG << "Cable Id =" << cableId;
+                auto respHandler = [asyncResp,
+                                    cableId](const boost::system::error_code ec,
+                                             ManagedPropertyType& resp) {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
+
+                    getCableProperties(asyncResp, resp, cableId);
+                };
+
+                crow::connections::systemBus->async_method_call(
+                    respHandler, "xyz.openbmc_project.Inventory.Manager",
+                    "/xyz/openbmc_project/inventory/cables/" + cableId,
+                    "org.freedesktop.DBus.Properties", "GetAll", "");
+            });
+
+    BMCWEB_ROUTE(app, "/redfish/v1/Cables/<str>/")
+        .privileges({{"Login"}})
```

Comment:
Same as line 57.

Reply:
Let us see what @gtmills says...

Reply:
I suspect patching a cable requires at least ConfigureComponents.  But I don't think we require patch support for cables anyway, so maybe we can skip patch support for now?